voidcrew/modules/cloning/code/cloning.dm

```diff
@@ -0,0 +1,600 @@
+	var/flesh_number = 0
+	///Boolean on whether to allow experimental cloning, allowing different minds in one body.
+	var/experimental_pod = FALSE
+
+	var/internal_radio = TRUE
+	var/obj/item/radio/radio
+	var/radio_key = /obj/item/encryptionkey/headset_med
+	var/radio_channel = RADIO_CHANNEL_MEDICAL
+
+/obj/machinery/clonepod/Initialize(mapload)
+	. = ..()
+	create_reagents(100, OPENCONTAINER)
+	countdown = new(src)
+	if(internal_radio)
+		radio = new(src)
+		radio.keyslot = new radio_key
+		radio.subspace_transmission = TRUE
+		radio.canhear_range = 0
+		radio.recalculateChannels()
+
+/obj/machinery/clonepod/Destroy()
+	var/mob/living/mob_occupant = occupant
+	go_out()
+	if(mob_occupant)
+		// Random comment: this is a bad situation since breaking the pod ejects the occupant
+		log_combat("[key_name(mob_occupant)] ejected from [src] at [AREACOORD(src)] due to Destroy().")
+	QDEL_NULL(radio)
+	QDEL_NULL(countdown)
+	if(connected)
+		connected.DetachCloner(src)
+	QDEL_LIST(unattached_flesh)
+	return ..()
+
+/obj/machinery/clonepod/RefreshParts()
+	. = ..()
+	speed_coeff = 0
+	efficiency = 0
+	if(reagents)
+		reagents.maximum_volume = 0
+	fleshamnt = 1
+	for(var/obj/item/reagent_containers/G in component_parts)
+		reagents.maximum_volume += G.volume
+		G.reagents.trans_to(src, G.reagents.total_volume)
+	for(var/obj/item/stock_parts/scanning_module/S in component_parts)
+		efficiency += S.rating
+		fleshamnt = 1/max(efficiency-1, 1)
+	for(var/obj/item/stock_parts/manipulator/P in component_parts)
+		speed_coeff += P.rating
+	heal_level = (efficiency * 15) + 10
+	if(heal_level < MINIMUM_HEAL_LEVEL)
+		heal_level = MINIMUM_HEAL_LEVEL
+	if(heal_level > 100)
+		heal_level = 100
+
+/obj/machinery/clonepod/attack_hand(mob/user)
+	. = ..()
+	if(.)
+		return
+	user.examinate(src)
+
+/obj/machinery/clonepod/AltClick(mob/user)
+	if(tgui_alert(user, "Are you sure you want to empty the cloning pod?", "Empty Reagent Storage:", "Yes", "No") != "Yes")
+		return
+	to_chat(user, span_notice("You empty \the [src]'s release valve onto the floor."))
+	src.reagents.clear_reagents()
+
+/obj/machinery/clonepod/attack_ai(mob/user)
+	return attack_hand(user)
+
+/obj/machinery/clonepod/examine(mob/user)
+	. = ..()
+	. += span_notice("The <i>linking</i> device can be <i>scanned<i> with a multitool. It can be emptied by Alt-Clicking it.")
+	if(in_range(user, src) || isobserver(user))
+		. += span_notice("The status display reads: Cloning speed at <b>[speed_coeff*50]%</b>.<br>Predicted amount of cellular damage: <b>[100-heal_level]%</b><br> Storing up to <b>[reagents.maximum_volume]cm<sup>3</sup></b> of synthflesh.")
+		. += span_notice("Synthflesh consumption at <b>[round(fleshamnt*90, 1)]cm<sup>3</sup></b> per clone.")
+		. += span_notice("The reagent display reads: [round(reagents.total_volume, 1)] / [reagents.maximum_volume] cm<sup>3</sup>")
+		if(efficiency > 5)
+			. += span_notice("Pod has been upgraded to support autoprocessing and apply beneficial mutations.")
```

Comment:
Do we really want people to get free mutations from killing and autocloning themselves??
I thought we had agreed to remove the feature entrely and only have it be "no negative mutations"

Reply:
Yeah i can remove the mutations from happening I think I forgot to remove them